wsagent/che-core-api-git-shared/src/main/java/org/eclipse/che/api/git/shared/ProviderInfo.java

```diff
@@ -28,6 +28,10 @@
 
     private Map<String, String> info = new HashMap<>();
 
+    public ProviderInfo(@NotNull String providerName) {
+        info.put(PROVIDER_NAME, providerName);
+    }
+
```

Comment:
also please fix java doc of getAuthenticateUrl method and describe that it can return null value and add org.eclipse.che.commons.annotation.Nullable for it